packages/application-shell/src/components/application-shell/application-shell.spec.js

```diff
@@ -1,508 +1,446 @@
-        wrapper = shallow(<RestrictedApplication {...props} />)
-          .find(FetchUser)
-          .renderProp('children')(userData)
-          .find(AsyncLocaleData)
-          .renderProp('children')({
-          isLoading: true,
-          locale: null,
-          messages: null,
-        });
-      });
-      it('should not pass "locale" prop to <ConfigureIntlProvider>', () => {
-        expect(wrapper.find(ConfigureIntlProvider)).not.toHaveProp('locale');
-      });
-      it('should not pass "messages" prop to <ConfigureIntlProvider>', () => {
-        expect(wrapper.find(ConfigureIntlProvider)).not.toHaveProp('messages');
-      });
+  });
+  it('should pass environment into application context', async () => {
+    const TestComponent = () => {
+      const applicationName = useApplicationContext(
+        context => context.environment.applicationName
+      );
+      return <p>{`Application name: ${applicationName}`}</p>;
+    };
+    const rendered = renderApp(<TestComponent />);
+    await waitForElement(() => rendered.getByText('Application name: my-app'));
+  });
+});
+describe('when route does not contain a project key', () => {
+  beforeEach(() => {
+    createGraphqlMockServer(xhrMock, {
+      operationsByTarget: { mc: mocksForMc.createMockOperations() },
     });
-    describe('when fetching the user returns an error', () => {
-      beforeEach(() => {
-        reportErrorToSentry.mockClear();
-        props = createTestProps();
-        userData = {
-          isLoading: false,
-          error: new Error('Failed to fetch'),
-        };
-        wrapper = renderForAsyncData({ props, userData });
-      });
-      it('should pass "locale" to <ConfigureIntlProvider>', () => {
-        expect(wrapper.find(ConfigureIntlProvider)).toHaveProp('locale', 'en');
-      });
-      it('should render <ErrorApologizer>', () => {
-        expect(wrapper).toRender('ErrorApologizer');
-      });
-      it('should report error to sentry', () => {
-        expect(reportErrorToSentry).toHaveBeenCalledWith(
-          new Error('Failed to fetch'),
-          {}
-        );
-      });
-      it('should pass default "messages" to <ConfigureIntlProvider>', () => {
-        expect(wrapper.find(ConfigureIntlProvider)).toHaveProp(
-          'messages',
-          expect.objectContaining({
-            'AppKit.title': 'Title en',
-          })
-        );
-      });
-      it('should pass application "messages" to <ConfigureIntlProvider>', () => {
-        expect(wrapper.find(ConfigureIntlProvider)).toHaveProp(
-          'messages',
-          expect.objectContaining({
-            'CustomApp.title': 'Title en',
-          })
-        );
-      });
-      it('should not render <ApplicationContextProvider>', () => {
-        expect(wrapper).not.toRender(ApplicationContextProvider);
-      });
+  });
+  it('should not render NavBar', async () => {
+    const rendered = renderApp();
+    await waitForElement(() => rendered.getByText('OK'));
```

Comment:
Nit I think we can use `await findByText` here everywhere to make it a bit leaner maybe?

Reply:
Hmm what do you mean?

Reply:
To my understandting RTL released and documented the `findBy`-APIs which are like what we have here so `await waitForElement(() => rendered.getByText('OK'));` can be shortenede to `await rendered.findByText('OK')`.

Reply:
Ah, I thought it was only used in cypress or something.

Sure, I we can do that, thanks for the heads up